qiime/core/type/signature.py

```diff
@@ -94,10 +110,30 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    @property
+    def descriptions(self):
+        params = {**self.inputs, **self.parameters, **self.outputs}
```

Comment:
You're converting the ordered keys from three OrderedDicts into a set, which is unordered. Right now it's misleading because an OrderedDict with an arbitrary order is returned. We should pick one of the options I listed, I don't have a strong opinion on either.